cmd/mdatagen/metrics_v2.tmpl

```diff
@@ -44,17 +44,33 @@ func DefaultMetricsSettings() MetricsSettings {
 	}
 }
 
+// metric holds data for generated metric and keeps track of data points slice capacity.
+type metric struct {
+	data     pdata.Metric // data buffer for generated metric.
+	capacity int          // max observed number of data points added to the metric.
+}
```

Comment:
I'm thinking of making this class metric specific and moving record and emit function to this class. I'll prepare another refactoring PR once this one is merged

Reply:
Emit func should probably be at the top level so it only needs to be called once. Unless we want to emit metrics one at a time, but this is not my expectation.

Reply:
Exported Emit method will stay the same, here I mean just refactoring of internal methods

Reply:
And actually I meant init not emit. updated commend

Reply:
Ah, that makes sense. Thanks for clarifying.